src/build_system.ml

```diff
@@ -1528,29 +1552,34 @@ let package_deps t pkg files =
         acc
       else begin
         rules_seen := Rule.Id.Set.add !rules_seen ir.id;
-        let _, dyn_deps =
-          match ir.exec with
-          | Running { rule_evaluation; _ }
-          | Evaluating_rule { rule_evaluation; _ } ->
-            Option.value_exn (Fiber.Future.peek rule_evaluation)
-          | Not_started _ -> assert false
-        in
-        let action_deps =
-          Static_deps.action_deps
-            (Lazy.force ir.static_deps)
-        in
-        Path.Set.fold
-          (Deps.path_union action_deps dyn_deps)
-          ~init:acc ~f:loop
+        match Fiber.Once.peek ir.static_deps with
+        | None ->
+          (* We know that at this point of execution, all the relevant
+             ivars have been filled *)
+          assert false
```

Comment:
Having `Fiber.Once.peek_exn` would make this code a little cleaner I think - a little less nesting to the right.

Reply:
Indeed, done